playbooks/files/promtail/promtail.yml.j2

```
@@ -0,0 +1,21 @@
+# vim:ts=2:sw=2:et:ai:sts=2
+---
+server:
+  http_listen_port: 9080
+  grpc_listen_port: 0
+
+positions:
+  filename: /var/lib/promtail/positions.yaml
+
+clients:
+  - url: {{ promtail__push_url }}
```

It would be good to be able to set more than one loki URL, and also to set extra parameters like authentication. The simplest would be to just accept a list of client definitions:

```suggestion
clients:
  {{ promtail__clients | to_nice_yaml(indent=2) | indent(2) }}
```

And to make the default case simple, you can set a default that creates config objects from a list of plain URLs:

```yaml
promtail__client_urls:
- http://host1:9000/

promtail__client_defaults:
  basic_authentication:
    username: foo

promtail__clients: |
  {% set res = [] %}
  {% for url in promtail__client_urls %}
  {% set _ = res.append(promtail__client_defaults | combine(dict(url=url))) %}
  {% endfor %}
  {{- res -}}
```

---

Applied the suggestion. But I have added `promtail__clients`  here in this PR (to generic playbook) and `promtail__client_urls` and `promtail__client_defaults` to hive PR (https://github.com/HiveNetCode/hive-compute-sysadmin/pull/107) since it is particular to the client. But maybe I should leave this default vars (with host11 and foo) as an example? PTAL.

---

Yes, that is exactly what I meant, but in you'll need to add some empty defaults, so the templates don't break easily. I will leave a comment